algorithms.py

```diff
@@ -679,10 +961,50 @@ def initialise(self, ts):
         for node in range(ts.num_nodes):
             seg = root_segments_head[node]
             if seg is not None:
+                left_end = seg.left
```

Comment:
Would it be easier to do this stuff as a separate loop through the nodes below? So, all the stuff we do during init is under the one "if model == "smc_k"?